model/src/main/kotlin/config/LicenseChoices.kt

```diff
@@ -29,11 +29,21 @@ import org.ossreviewtoolkit.spdx.model.LicenseChoice
  * The license choices configured for a repository.
  */
 data class LicenseChoices(
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val repositoryLicenseChoices: List<LicenseChoice> = emptyList(),
+
     @JsonInclude(JsonInclude.Include.NON_EMPTY)
     val packageLicenseChoices: List<PackageLicenseChoice> = emptyList()
 ) {
     @JsonIgnore
-    fun isEmpty() = packageLicenseChoices.isEmpty()
+    fun isEmpty() = packageLicenseChoices.isEmpty() && repositoryLicenseChoices.isEmpty()
+
+    init {
+        val choicesWithoutGiven = repositoryLicenseChoices.filter { it.given == null }
+        require(choicesWithoutGiven.isEmpty()) {
```

Comment:
Why do `repositoryLicenseChoices` require a `given` but `packageLicenseChoices` do not?

Reply:
When using a package license choice we have a specific effective license of the package a choice (or several) can be applied to, which is most likely to be known before applying the choice. When we have a repository license choice, the license choice would be applied to each package that offers this license as a choice. Making sure that there is a given helps only applying the choice to a wanted given as opposed to all licenses with that choice, which could lead to unwanted choices.
For consistency it would be possible to make `given` mandatory for `packageLicenseChoices`, but this was previously discussed before and decided otherwise.

Reply:
Thanks for the explanation, can you add that to the code docs, e.g. for the `LicenseChoices` class or the properties? And maybe also to the documentation from the last commit?

Reply:
I added the explanation to the `RepositoryLicenseChoices`, but am unsure if this is how and where it belongs. 
I also added it to the documentation.

Reply:
Thanks, I think this is good enough for now. I just wanted to avoid that when you read this code without knowing the context you stumble over the fact that only one of the properties has this requirement.